internal/hardware/hardware.go

```diff
@@ -111,7 +106,26 @@ func (s *HardwareInfo) GetMutableHardwareInfoDelta(hardwareMutableInfoPrevious m
 }
 
 func (s *HardwareInfo) getHostDevices() ([]*models.HostDevice, error) {
-	return getDevices("/dev")
+	runcDevices, err := runc.HostDevices()
+	if err != nil {
+		return []*models.HostDevice{}, err
+	}
+
+	devices := make([]*models.HostDevice, 0, len(runcDevices))
+	for _, d := range runcDevices {
+		device := models.HostDevice{
+			Path:       d.Path,
+			DeviceType: string(d.Type),
+			Major:      d.Major,
+			Minor:      d.Minor,
+			UID:        int64(d.Uid),
+			Gid:        int64(d.Gid),
+		}
+
+		devices = append(devices, &device)
+	}
```

Comment:
Could this be extracted into a private method `extractRuncDevices` (or something similar) or is that too much for you? (I tend to refactor as much as possible to have as small methods/functions as possible)

Reply:
I don't think it adds value. It's just a mapping.